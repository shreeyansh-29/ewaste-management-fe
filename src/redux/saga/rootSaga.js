import {all} from "redux-saga/effects";
import {watchCustomerName} from "./customer/customerNameSaga/customerNameSaga";
import {watchCustomerNotificationCount} from "./customer/customerNotificationSaga/customerNotificationCountSaga";
import {watchCollectorNotificationData} from "./collector/collectorNotificationSaga/collectorNotificationDataSaga";
import {watchForgotPasswordSaga} from "./forgotPasswordSaga/forgotPasswordSaga";
import {watchResetPasswordSaga} from "./resetPasswordSaga/resetPasswordSaga";
import {watchSignInSaga} from "./signInSaga/signInSaga";
import {watchSignUpSaga} from "./signUpSaga/signUpSaga";
import {watchCollectorName} from "./collector/collectorNameSaga/collectorNameSaga";
import {watchVendorName} from "./vendor/vendorNameSaga/vendorNameSaga";
import {watchCollectorNotificationCount} from "./collector/collectorNotificationSaga/collectorNotificationCountSaga";
import {watchCustomerNotificationData} from "./customer/customerNotificationSaga/customerNotificationDataSaga";
import {watchCustomerProfile} from "./customer/customerProfileSaga/customerProfileSaga";
import {watchVendorProfile} from "./vendor/vendorProfileSaga/vendorProfileSaga";
import {watchCollectorProfile} from "./collector/collectorProfileSaga/collectorProfileSaga";
import {watchCustomerEWaste} from "./customer/customerEWasteSaga/customerEWasteSaga";
import {watchCustomerDropOff} from "./customer/customerDropOffSaga/customerDropOffSaga";
import {watchCustomerViewCollectors} from "./customer/customerViewCollectorSaga/customerViewCollectorSaga";
import {watchCustomerWasteGenerated} from "./customer/analyticsSaga/customerWasteGeneratedSaga";
import {watchCustomerCollectorCategories} from "./customer/analyticsSaga/customerCollectorCategoriesSaga";
import {watchCustomerDrives} from "./customer/analyticsSaga/customerDrivesSaga";
import {watchCollectorEWasteDrives} from "./collector/analyticsSaga/collectorEWasteDrivesSaga";
import {watchCollectorEWasteItems} from "./collector/analyticsSaga/collectorEWasteItemsSaga";
import {watchCollectorRevenueChart} from "./collector/analyticsSaga/collectorRevenueChartSaga";
import {watchCollectorUsers} from "./collector/analyticsSaga/collectorUsersSaga";
import {watchCollectorVendor} from "./collector/analyticsSaga/collectorVendorSaga";
import {watchVendorCategory} from "./vendor/analyticsSaga/vendorCategorySaga";
import {watchVendorCatgItems} from "./vendor/analyticsSaga/vendorCatgItemsSaga";
import {watchVendorVendorData} from "./vendor/analyticsSaga/vendorVendorDataSaga";
import {watchVendorCollectorData} from "./vendor/analyticsSaga/vendorCollectorDataSaga";
import {watchVendorViewItems} from "./vendor/vendorSalesSaga/vendorViewItemsSaga";
import {watchVendorAcceptItems} from "./vendor/vendorSalesSaga/vendorAcceptItemsSaga";
import {watchVendorMyOrders} from "./vendor/vendorMyOrdersSaga/vendorMyOrdersSaga";
import {watchCollectorOrganizeDrive} from "./collector/collectorOrganizeDriveSaga/collectorOrganizeDriveSaga";
import {watchCollectorForSale} from "./collector/collectorForSaleSaga/collectorForSaleSaga";
import {watchCollectorMyDrives} from "./collector/collectorMyDrivesSaga/collectorMyDrivesSaga";
import {watchCollectorMyDrivesStatus} from "./collector/collectorMyDrivesStatusSaga/collectorMyDrivesStatusSaga";
import {watchCollectorAvailable} from "./collector/collectorAvailableSaga/collectorAvailableSaga";
import {watchCollectorSold} from "./collector/collectorSoldSaga/collectorSoldSaga";
import {watchViewAcceptColProfileSaga} from "./vendor/viewAcceptColProfileSaga";
import {watchViewCollProfileSaga} from "./vendor/viewCollProfileSaga";
import {watchViewCollectorProfileSaga} from "./customer/viewCollectorProfileSaga";
import {watchVendorProfileEdit} from "./vendor/vendorProfileSaga/vendorProfileEditSaga";
import {watchCustomerProfileEdit} from "./customer/customerProfileSaga/customerProfileEditSaga";
import {watchCollectorProfileEdit} from "./collector/collectorProfileSaga/collectorProfileEditSaga";
import {watchCustomerCountColl} from "./customer/customerPickUpSaga/customerCountCollSaga";
import {watchCustomerPickUp} from "./customer/customerPickUpSaga/customerPickUpSaga";
import {watchCollectorPending} from "./collector/collectorPendingSaga/collectorPendingSaga";
import {watchCollectorPendingAccept} from "./collector/collectorPendingAcceptSaga/collectorPendingAcceptSaga";
import {watchCollectorSummary} from "./collector/collectorSummarySaga/collectorSummarySaga";
import {watchCustomerCompletedRequest} from "./customer/customerCompletedRequestSaga/customerCompletedRequestSaga";
import {watchCustomerPendingRequest} from "./customer/customerPendingRequestSaga/customerPendingRequestSaga";
import {watchCustomerExpiredRequest} from "./customer/customerExpiredRequestSaga/customerExpiredRequestSaga";
import {watchCustomerPendingDecline} from "./customer/customerPendingRequestSaga/customerPendingDeclineSaga";

function* rootSaga() {
  yield all([
    watchForgotPasswordSaga(),
    watchResetPasswordSaga(),
    watchSignInSaga(),
    watchSignUpSaga(),
    watchCustomerName(),
    watchCustomerNotificationCount(),
    watchCustomerNotificationData(),
    watchCustomerProfile(),
    watchCustomerProfileEdit(),
    watchCustomerEWaste(),
    watchCustomerPickUp(),
    watchCustomerCountColl(),
    watchCustomerDropOff(),
    watchCustomerViewCollectors(),
    watchCustomerWasteGenerated(),
    watchCustomerCollectorCategories(),
    watchCustomerDrives(),
    watchCustomerCompletedRequest(),
    watchCustomerPendingRequest(),
    watchCustomerExpiredRequest(),
    watchCustomerPendingDecline(),
    watchCollectorName(),
    watchCollectorNotificationCount(),
    watchCollectorNotificationData(),
    watchCollectorProfile(),
    watchCollectorProfileEdit(),
    watchCollectorEWasteDrives(),
    watchCollectorEWasteItems(),
    watchCollectorRevenueChart(),
    watchCollectorUsers(),
    watchCollectorVendor(),
    watchCollectorOrganizeDrive(),
    watchCollectorForSale(),
    watchCollectorAvailable(),
    watchCollectorSold(),
    watchCollectorMyDrives(),
    watchCollectorMyDrivesStatus(),
    watchCollectorPending(),
    watchCollectorPendingAccept(),
    watchCollectorSummary(),
    watchVendorName(),
    watchVendorProfile(),
    watchVendorProfileEdit(),
    watchVendorCatgItems(),
    watchVendorCategory(),
    watchVendorCollectorData(),
    watchVendorVendorData(),
    watchVendorMyOrders(),
    watchVendorViewItems(),
    watchVendorAcceptItems(),
    watchViewAcceptColProfileSaga(),
    watchViewCollProfileSaga(),
    watchViewCollectorProfileSaga(),
  ]);
}

export default rootSaga;
