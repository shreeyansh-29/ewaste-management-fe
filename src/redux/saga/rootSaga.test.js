/* eslint-disable no-undef */
import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import rootSaga from "./rootSaga";
import toJson from "enzyme-to-json";

Enzyme.configure({adapter: new Adapter()});

describe("test rootSaga", () => {
  let RootSaga = rootSaga;
  const wrapper = shallow(<RootSaga />);

  it("testing rootSaga", () => {
    expect.assertions(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
