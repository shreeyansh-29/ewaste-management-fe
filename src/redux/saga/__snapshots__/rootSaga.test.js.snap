// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test rootSaga testing rootSaga 1`] = `<Component />`;
